### PR TITLE
Fix social signup

### PR DESCRIPTION
**Pull request by Tug**
This fixes social login by re-enabling the popup flow for Chrome.
It also has commits from https://github.com/Automattic/wp-calypso/pull/18712 which did not actually introduce any bug but was reverted.

### Testing Instructions

- Open an incognito session in google chrome
- Boot https://calypso.live/?branch=fix/social-signup-redirect
 go to https://woocommerce.com/ and click on `SIGN IN WITH WORDPRESS.COM`
- Replace `wordpress.com` in your url by `calypso.live`
- Click on the `Continue with Google` button
- Assert that it uses the popup flow and that you can sign-in/sign-up as usual with your google account
- Open Safari and navigate to https://calypso.live/?branch=fix/social-signup-redirect click on Log In
- click on `SIGN IN WITH WORDPRESS.COM`
- Assert that it uses the redirect flow and that you can log in
- Log out and navigate to https://woocommerce.com/
- click on `SIGN IN WITH WORDPRESS.COM`
- Replace `wordpress.com` in your url by `calypso.live`
- Click on the `Continue with Google` button
- Assert that it uses the redirect flow and that you can log in and authorize woo (or continue the flow)

### Reviews
- [ ] Code
- [ ] Product

### CODE REVIEW

File: client/blocks/login/social.jsx [hunk 1]
```diff
@@ -28,13 +28,15 @@ import { InfoNotice } from 'blocks/global-notice';
 import { login } from 'lib/paths';
 
 function isSafari() {
-	return typeof window !== 'undefined' && window.navigator.userAgent.match( /safari/i );
+	return (
+		typeof window !== 'undefined' && /^(?!.*chrome).*safari/i.test( window.navigator.userAgent )
```
Comment (yurynix) [inline client/blocks/login/social.jsx]: Why chrome is in the regex?
Safari has something like:
`Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Safari/604.1.38`

Comment (yurynix) [reply client/blocks/login/social.jsx]: oops, I missed the `?!`, sorry =)

Comment (Tug) [reply client/blocks/login/social.jsx]: Try `navigator.userAgent` on chrome you'll see ;)
In my case I have this:
`"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36"`

File: client/blocks/login/social.jsx [hunk 2]
```diff
@@ -28,13 +28,15 @@ import { InfoNotice } from 'blocks/global-notice';
...
+	);
 }
 
 function shouldUseRedirectFlow() {
 	// If calypso is loaded in a popup, we don't want to open a second popup for social login
 	// let's use the redirect flow instead in that case
-	const isPopup = typeof window !== 'undefined' && window.opener && window.opener !== window;
+	const isPopup = typeof window !== 'undefined' && window.opener !== window;
```
Comment (yurynix) [inline client/blocks/login/social.jsx]: screen shot 2017-10-11 at 0 55 28

I think it should remain `&& window.opener &&`

File: client/blocks/login/social.jsx (hunk 1 above)
Comment (yurynix) [reply client/blocks/login/social.jsx]: Yeah, I haven't noticed the lookahead, sorry =)

File: client/blocks/login/social.jsx (hunk 2 above)
Comment (Tug) [reply client/blocks/login/social.jsx]: Right 👍

**Comment by Tug** [comment]
@yurynix Finally found out why it wasn't working as expected on calypso.live. I believe it's because 3rd party cookies are disabled by default on this browser.
You can test the same flow with https://calypso.live/?branch=master and you'll have the same results, doesn't matter if we use the popup or the redirect flow (I added a commit to force the popup flow for oauth2 client login)

**Pull request merged**